### PR TITLE
Test List Endpoint Query in Mosaic Production

### PR DESCRIPTION
**Pull request by Renny4Real**
This commit adds changes found while testing the Mosaic production API:
- Creates a dbo schema for the test database to match structure of production DB.
- Changes data types & lengths in infrastucture columns to match the ones expected in the production DB.
- Refactors the get all residents gateway method so that querying is quicker and returns distinct data.
- Adds missing Mosaic Id in response object.

To Do: In a separate PR, will add missing phone types that are present in the DB but have not be added as enum values in the code base.

### CODE REVIEW

File: MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs [hunk 1]
```diff
@@ -15,35 +20,114 @@ public class MosaicGateway : IMosaicGateway
 
         public MosaicGateway(MosaicContext mosaicContext)
         {
+            Console.WriteLine("In gateway constructor");
```
Comment (elcorbs) [inline MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs:23]: Leftover comment

File: MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs [hunk 2]
```diff
@@ -15,35 +20,114 @@ public class MosaicGateway : IMosaicGateway
...
+            {
+                AddressLine1 = a.AddressLines,
+                PostCode = a.PostCode
+            }).Distinct().ToList();
+
+            return !addressList.Any() ? null : addressList;
         }
 
-        public ResidentInformation GetEntityById(int id)
+        public ResidentInformation GetEntityById(long id)
```
Comment (elcorbs) [inline MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs:130]: Can we move the public method to be above the private ones please

File: MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs [hunk 3]
```diff
@@ -15,35 +20,114 @@ public class MosaicGateway : IMosaicGateway
...
+                            string.IsNullOrEmpty(address) || EF.Functions.ILike(add.AddressLines.Replace(" ", ""), addressSearchPattern))
+                        .Where(add =>
+                            string.IsNullOrEmpty(postcode) || EF.Functions.ILike(add.PostCode.Replace(" ", ""), postcodeSearchPattern))
+                        .Distinct()
+                        .ToList(),
+                    TelephoneNumbers = _mosaicContext.TelephoneNumbers.Where(n => n.PersonId == p.Id).Distinct().ToList()
+                }).ToList();
+
+            return dbRecords.Select(x => new ResidentInformation
+            {
```
Comment (elcorbs) [inline MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs:57]: I think we can use the method `MapPersonAndAddressesToResidentInformation ` here

File: MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs (hunk 1 above)
Comment (Renny4Real) [reply MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs:23]: Thanks 👍

File: MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs (hunk 2 above)
Comment (Renny4Real) [reply MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs:130]: Sure, will do that.

File: MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs (hunk 3 above)
Comment (Renny4Real) [reply MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs:57]: Yep. Nice! 👍

Comment (Renny4Real) [reply MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs]: Added in d454b7a

File: MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs (hunk 2 above)
Comment (Renny4Real) [reply MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs]: Done in d454b7a

File: MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs (hunk 1 above)
Comment (Renny4Real) [reply MosaicResidentInformationApi/V1/Gateways/MosaicGateway.cs]: Removed in d454b7a

**Pull request merged**